flocker/control/_model.py

```diff
@@ -356,6 +357,17 @@ class DatasetChanges(object):
     """
 
 
+class _PathMap(CheckedPMap):
+    """
+    A mapping between dataset IDs and the paths where they are mounted.
+
+    See https://github.com/tobgu/pyrsistent/issues/26 for more succinct
+    idiom combining this with ``field()``.
+    """
+    __key_type__ = unicode
+    __value_type__ = FilePath
```

Comment:
Kind of unrelated, but I wonder if there's a way to make these interface-based instead of type-based.